src/stream/tests/integration_tests/snapshot.rs

```diff
@@ -198,7 +199,7 @@ fn run_until_pending(
                 }
                 SnapshotEvent::Chunk(output)
             }
-            Message::Barrier(barrier) => SnapshotEvent::Barrier(barrier.epoch.curr),
+            Message::Barrier(barrier) => SnapshotEvent::Barrier(barrier.epoch.curr / test_epoch(1)),
```

Comment:
The changes here are subtle given that the input/output of `check_with_script` are epochs without shift. It is better to add some comments here to explain why we need the changes.